welder.py

```diff
@@ -0,0 +1,52 @@
+import subprocess, os
+from flask import (Flask, request, abort, jsonify)
+
+app = Flask(__name__)  # create the application instance
+
+
+@app.route('/api/syncTo', methods=['POST'])
+def sync_to():
+    if not request or 'destination' not in request.json:
+        return abort(400)
+    destination = request.json['destination']
+    source = os.getenv('WELDERVOLUME','')
+    if not validate_bucket(destination):
+        return jsonify({'Destination is not a valid bucket':destination}), 400
+    if (source == ''):
+        return 'Source not defined', 400
+    return run_command(source, destination)
+
+
+@app.route('/api/syncFrom', methods=['POST'])
+def sync_from():
+    if not request or 'source' not in request.json:
+        return abort(400)
+    source = request.json['source']
+    destination = os.getenv('WELDERVOLUME','')
+    if not validate_bucket(source):
+        return 'Source {0} is not a valid GCS bucket'.format(source), 400
+    if (destination == ''):
+        return 'Destination not defined', 400
+    return run_command(source, destination)
+
+
+def validate_bucket(bucket_name):
+    return bucket_name.startswith('gs://')
+
+
+def run_command(source, destination):
+    command = ['gsutil', '-m', 'rsync', '-r', source, destination]
+    output = ''
+    p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    output= list((iter(p.stdout.readline, b'')))
+    if('exception' in output[-2].decode('utf-8').lower()):
+        return jsonify({'status':'Error', 'message':output[-2].decode('utf-8').strip()}), 403
+    else:
+        if('Starting synchronization...' in output[-1].decode('utf-8').lower()):
+            return jsonify({'status':'Success', 'message':'Files are up to date.'}), 200
```

Comment:
having `status: Success` in the response body seems redundant. i would prefer that the response codes are the sole indicator of the request succeeding. 

we should also think about the shape of this response and if it's right. 

Maybe a 200 OK response looks something like:

```
{
  "succeeded": 271,
  "failed": 0,
  "totalFiles": 271
}
```

Reply:
fwiw i suggested the `status` key, though my opinion isn't super strong. i liked the idea of clients being able to use the same deserializer for both kinds of responses. i think cromwell does a similar thing, though i can't really speak to their reasoning. i abstain my vote here